lib/vehicle_factory.rb

```diff
@@ -0,0 +1,107 @@
+class VehicleFactory
+attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(api)
+  # Helper method here
+    sort_out_boats_and_trl(api)
+    api.each do |vehicle|
+      if vehicle[:vin].nil?
+        vehicle[:vin] = vehicle.delete :vin_1_10
+      end
+      if vehicle[:model] == nil
+        vehicle[:model] = "Model not listed"
+      end
+      if vehicle[:fuel_type].nil? 
+        vehicle[:engine] = :ev
+      else
+        vehicle[:engine] = vehicle[:fuel_type]
+      end
+      vehicle[:year] = vehicle.delete :model_year
+      vehicle = Vehicle.new(vehicle)
+      @vehicles << vehicle
+    end
+  end
+
+  def most_popular_ev
+    ev_counter = Hash.new(0)
+    make_and_models.values.flatten.each do |value|
+    # value, or the model, becomes a key for the ev_counter
+      ev_counter[value] += 1
+    end
+    make = nil
+    # The model variable finds the most popular model
+    model = ev_counter.key(ev_counter.values.sort[-1])
+    # Now iterate through the make_and_models to find the make of the model
+    make_and_models.each do |key, value|
+      if value.include?(model)
+        make = key
+      end
+    end
+    "The most popular make and model is the #{make} #{model}!"
+  end
```

Comment:
Great job with these analytics. Could also create an analytics class and have these be methods in that class. Remember that even classes should adhere to Single Responsibility Principle.